### PR TITLE
Remove accessible menus

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Ticket:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/8717

**Related comment:** https://github.com/department-of-veterans-affairs/vets-website/pull/12463#discussion_r419500923

Removes `accessible-menus.js` and related files.

## Testing done
Ensuring tests pass CICD.

## Screenshots
N/A

## Acceptance criteria
- [x] Remove `accessible-menus.js` and related files.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/index.js [hunk 1]
```diff
@@ -25,10 +24,6 @@ import '@department-of-veterans-affairs/formation/dist/formation';
  * @param {Store} commonStore The Redux store being used by this application
  */
 export default function startSitewideComponents(commonStore) {
-  if (document.querySelector('#vetnav-menu') !== null) {
-    addMenuListeners(document.querySelector('#vetnav-menu'), true);
-  }
```
Comment (ncksllvn) [inline src/platform/site-wide/index.js:30]: This is so confusing to me...we're using a React menu so I can't imagine we need an additional set of event listeners applied via `addMenuListeners`, but `#vetnav-menu'` _does_ exist in the DOM tree, so does that we mean _do_ need both (React component and listeners) or are we just needlessly applying this module by mistake?

CC: @department-of-veterans-affairs/frontend-review-group

**Comment by kelsonic** [comment]
@department-of-veterans-affairs/frontend-review-group we could use some 👀and insight into this PR to make sure we're not breaking anything by removing this code 🙂

File: src/platform/site-wide/index.js (hunk 1 above)
Comment (U-DON) [reply src/platform/site-wide/index.js:30]: I think this is fine... The `addMenuListeners` doesn't add any listeners if the element passed in doesn't have `role="menu"` or `role="menubar"`, and it doesn't seem like `#vetnav-menu` has either.

**Review by ncksllvn — APPROVED** [review]

**Comment by kelsonic** [comment]
@department-of-veterans-affairs/frontend-review-group Ready for review! 🎉

**Review by theodur — APPROVED** [review]
Because `#vetnav-menu` doesn't have any attributes, `addMenuListeners` isn't doing anything do it so this looks good to me.

**Pull request merged**